cirq-google/cirq_google/calibration/engine_simulator.py

```diff
@@ -304,6 +305,45 @@ def create_from_characterizations_sqrt_iswap(
             ideal_when_missing_parameter=ideal_when_missing_parameter,
         )
 
+    @classmethod
+    def create_from_gate_dict(
+        cls,
+        gates_dict: Dict[
+            Tuple[cirq.Qid, cirq.Qid], Dict[cirq.FSimGate, PhasedFSimCharacterization]
+        ],
+    ) -> 'PhasedFSimEngineSimulator':
+        """Creates PhasedFSimEngineSimulator with fixed drifts.
+
+        Args:
+            gates_dict: maps every pair of qubits and engine gate on that pair to a
+                characterization for that gate.
+
+        Returns:
+            New PhasedFSimEngineSimulator instance.
+        """
+
+        def sample_gate(
+            a: cirq.Qid, b: cirq.Qid, gate: cirq.FSimGate
+        ) -> PhasedFSimCharacterization:
+            params = None
+            swapped = False
+            if (a, b) in gates_dict:
+                params = gates_dict[(a, b)].get(gate)
+            elif (b, a) in gates_dict:
+                params = gates_dict[(b, a)].get(gate)
+                swapped = True
+            if params is None:
+                raise ValueError(f'Missing params for value for pair {(a, b)} and gate {gate}.')
+            assert isinstance(params, PhasedFSimCharacterization)
+            if swapped:
+                params = params.parameters_for_qubits_swapped()
+            return params
+
+        simulator = cirq.Simulator()
```

Comment:
The original method had a check here which tested if all the pairs are sorted. This prevents providing the incorrect input where swapped gate is very different from the non-swapped gate. I think either one of those checks should be here, to prevent incorrect simulator state.

Reply:
Added the check.